src/utils/DownloadUtils.ts

```diff
@@ -21,4 +28,66 @@ export function addSolutionsToError(err: R2Error): void {
 export function generateProgressPercentage(progress: number, currentIndex: number, total: number): number {
     const completedProgress = (currentIndex / total) * 100;
     return completedProgress + (progress * 1/total);
-}
\ No newline at end of file
+}
+
+
+export async function getFullDependencyList(
+    profile: ImmutableProfile,
+    combos: ThunderstoreCombo[],
+): Promise<ThunderstoreCombo[]> {
+    const modList = await ProfileModList.getModList(profile);
+    if (modList instanceof R2Error) {
+        throw modList;
+    }
+
+    const allModsToDownload: ThunderstoreCombo[] = [...combos];
+
+    for (const combo of combos) {
+        const isModpack = combo.getMod().getCategories().find(value => value === "Modpacks") !== undefined;
+        const versionMode = isModpack
```

Comment:
Are you certain we can determine the builder mode internally in this method without external arguments and still make this a one-size-fits-all solution? Consider profile imports vs. update all vs. single mod install.

If it can be done (and might make sense even if not), then maybe `DependencySetBuilderMode` should reside in the same file instead of being imported from the download provider.

Reply:
Do you think this comment is resolved?

Reply:
The second part is resolved.
For the first part: At least to my testing, the builder works correctly, but unit testing could prove me wrong.

The whole enum could even be omitted, **IF** it turns out the mode only needs to be used once (or twice) and therefore is just same as checking `isModPack`.